### DISCUSSION TITLE
[시즌3] 모아보기 기능(모의면접 대비)

### DISCUSSION THREAD

**Discussion by gracefulBrown** [woowacourse/prolog#221] [Feature]
모아 보기 형태에 대해서 의견을 남겨주세요 :)

**Comment by gracefulBrown** [comment]
# Summary Report 기능
> ## 목적
> - 복습: 작성자는 기간별(미션별) 작성했던 학습로그를 다시보며 정리할 수 있음
> - 요약 이력 남기기: 요약된 레포트는 작성 후 포스트 처럼 관리 되어 이력처럼 남길 수 있음

## 리포트 작성
### 학습로그 선택
- 프로필 > 내가 쓴 포스트 목록 페이지로 이동
- 요약할 학습로그를 선택하여 요약의 대상 학습로그를 선정한다.
- 미션별 필터를 이용하면 쉽게 선택이 가능하다.

### 리포트 꾸미기
- 여러 개의 포스트로 작성되어있는 학습로그를 하나의 페이지로 요약한다
- 태그별로 키워드를 뽑아서 해당 키워드에 관련된 학습로그 내용을 한 곳에 모아서 작성한다.
- 주제별로 내용을 모을 수 있고 주제별 가중치를 부여할 수 있다.(선택)
- 선택된 학습로그들을 바탕으로 차트나 시각화 기능도 함께 첨부가 가능하다.(선택)

## 리포트 관리(CRUD)
- 작성했던 리포트는 리포트 별로 따로 관리를 할 수 있다.
- 미션별로 작성할 수 있고 낱개로 선택한 학습로그를 바탕으로 작성할 수 도 있다.

## 기대 효과
- 학습로그는 러프하게 작성했었더라면 리포트는 형식에 작성하여 (시각화를 하기에)학습로그 만으로는 부족한 정보를 보충해주는 역할을 기대할 수 있다.

**Comment by seovalue** [comment]
## Summary Report 기능

#### 목적
* 면접관에게 제공할 용도의 면접용 학습로그를 만들기 (작성한 학습로그 모으기)

## 면접용 학습로그 모아보기
여담을 약간 포함하자면, 저는 이전에 PR 마다 학습로그를 작성하고, 해당 학습로그를 gist를 이용해 모은 뒤 면접관에게 제공했어요.
제가 작성한 예시입니당 (https://gist.github.com/seovalue/ae7fac6e750f39f93b154523cb862e56)
하지만 PR에 달린 학습로그를 일일히 긁어서 하나로 합치는게 불편해서 **모을 수 있는 기능**이 프롤로그에 있으면 좋겠다고 생각했어요. 
브라운이 위에 남겨주신 학습로그 선택과 유사하지만 결이 다를수도 있을 것 같아요!

우선은 제안이라 제가 생각하는 흐름만 남겨놓을게요!
1. 학습로그를 작성한다. (아래 예시에서는 test, test1 포스팅)
2. 마이페이지 > Posts에 select 버튼이 존재
3. select 버튼으로 n개를 체크한다.
4. [학습로그 모으기(가칭)]라는 버튼이 존재해서 그 버튼을 누르면, 선택된 학습로그의 내용이 줄바꿈 한줄 포함 합쳐진 새로운 포스팅이 생긴다.

<img width="1015" alt="스크린샷 2021-08-10 오후 12 01 01" src="https://user-images.githubusercontent.com/48412963/128801790-99f797db-d4ee-4855-a990-f0fb97aff671.png">

**Reply by sunhpark42** [reply]
직접 학습로그를 택하는 것에 추가로, 특정 태그, 특정 과제 별 모으기가 가능하면 좋겠네용!

**Comment by devhyun637** [comment]
## Summary Report 기능
* 목적 :  
  - 스스로 학습한 내용을 키워드로 분류한 다음 나만의 학습 로드맵을 만들어간다.
  - 모의면접 뿐만 아니라, 본인을 포함한 다른 사람들이 한눈에 어떤 역량을 키우는 데 집중 했는지 파악할 수 있다.

* 기능 : 
  - 학습로그를 쓰기만 하면 만들어지는 `시각화 된 학습로그`
  - 역량을 시각화한다.


저는 개인적으로 크루들이 학습로그를 쓰고나서 이에 부과적인 활동(모의 면접을 위해서)을 해야한다면, 더 의욕을 떨어트리지 않을까? 하는 생각이 들었습니다. 그래서 `크루들은 쓰기만해! 우리가 예쁘게 정리해줄게`라는 컨셉으로 `스스로 학습한 내용을 키워드로 분류한 다음 나만의 학습 로드맵을 만들어간다는 방향성`을 생각하며 다음과 같은 디자인 시안을 만들어보았습니다. 또, 역량을 시각화해서 보여준다면 모의면접 뿐만 아니라, 실제로 본인 그리고 코치님들이 어떤 역량을 키우는 데 집중 했는지 한눈에 파악할 수 있지 않을까 하는 생각이 들었습니다.

(디자인 시안의 역량은 `프역셀체`에서 참고하였습니다.)

### 로직
1. 학습로그를 작성할 때, 대분류 된 역량을 한 개 이상 체크한다. (ex. Java, JavaScript & HTML, UX/UI 등)
2. 마이페이지의 OverView 항목에 가면 학습로그에 따라서 역량 그래프와 해시태그의 워드 클라우드 또는 해시태그 목록을 볼 수 있다.
3. 역량그래프의 한가지 영역을 선택하면 해당하는 워드클라우드(또는 해시태그 목록)와  이에 해당하는 학습로그 목록을 볼 수 있다.
4. 만약 대표적으로 보여주고 싶은 학습로그가 있다면, 대표 학습로그 등록 기능을 통해서 확인할 수 있다.

### 디자인 시안
디자인 1. 역량 그래프 + 워드 클라우드 + 테두리가 없는 학습로그 목록 / (왼쪽은 무시해주세요)
<img width="1128" alt="스크린샷 2021-08-16 오전 5 00 45" src="https://user-images.githubusercontent.com/59258239/129491046-601f9cbc-f424-4154-9110-dcb6b0200eac.png">

디자인 2. 역량 그래프 + 워드 클라우드 + 카드 형식 학습로그 목록
<img width="1113" alt="스크린샷 2021-08-16 오전 5 00 37" src="https://user-images.githubusercontent.com/59258239/129491040-010876bd-cea2-421e-9595-8bca9e6517aa.png">

디자인 3. 역량 그래프 + 워드 클라우드 + 테두리가 있는 학습로그 목록 / (왼쪽은 무시해주세요)
<img width="1145" alt="스크린샷 2021-08-16 오전 5 01 05" src="https://user-images.githubusercontent.com/59258239/129491055-530d0fce-55ca-4323-905f-74876779bd12.png">

디자인 4. 역량 그래프 + 해시태그 + 학습로그 목록
<img width="1159" alt="스크린샷 2021-08-16 오전 5 01 14" src="https://user-images.githubusercontent.com/59258239/129491060-7d5f36a5-2aee-4a29-99d1-80e56c42097a.png">

디자인 5. 역량 그래프 + 워드 클라우드 + 달력 + 대표 학습로그 + 학습로그 목록
<img width="1141" alt="스크린샷 2021-08-16 오전 5 00 55" src="https://user-images.githubusercontent.com/59258239/129491052-0a72bcab-bfaf-407e-bb12-db3748084eb7.png">

**Reply by gracefulBrown** [reply]
너무 멋진 계획이에요!
제가 볼 땐 리포트 기능쪽 보다는 오버뷰 시각화와 조금 더 관련이 있어 보이는데 회의할 때 논의해봐요~

**Comment by sunhpark42** [comment]
level1, level2 와 관련된 학습로그의 경우에는 프롤로그 측에서 레포트를 만들 때 과제별로 모아서 정리 및 가장 앞에 과제에 대한 설명을 간략히 추가해주는 것도 좋을 것 같습니다~
<예시>
```
[자동차 경주 게임] // 대제목 (=과제 명)
~ 자동차 경주 게임 설명
~ 관련 학습로그 1
~ 관련 학습로그 2
...
```

**Reply by gracefulBrown** [reply]
레(리)포트의 설명이 필요하다는 말씀이시죠? 
기본으로 제공되는 것 이외에 추가로 전달하고 싶은 메시지가 있을 수 있으니 꼭 필요할 꺼 같네요 👍

**Comment by sunhpark42** [comment]
## 2021.08.18 결정내용

- 리포트는 레벨 1 리포트, 레벨 2 등 리포트 목록으로 표시 될 수 있다.
- 리포트에는 역량그래프 및, 각 역량에 포함된 글 목록 조회가 가능하다.
    - 백엔드 : 역량 그래프, 글 조회 필터에 역량을 추가한다.
- 리포트 작성 시 역량 그래프에 가중치를 설정한다. 가중치는 100% 만점 기준으로 사용자가 스스로 결정한다.
- 리포트 작성 시 역량 그래프 중 표시 여부를 선택할 수 있다.
- 작성된 리포트를 삭제할 수 있다.
- 리포트 작성 시 역량을 선택한 후 글 추가를 할 때, 레벨별로 글을 필터링 할 수 있다.
    - 검색결과는 페이지네이션이 가능하다.
- 리포트 작성 시 제목과, 간단한 소개를 입력할 수 있다.
    - 제목은 default 작성일자
    - 간단한 소개는 optional
- 역량 관리 시에는 역량 이름 및 색상을 입력하여 생성할 수 있다.
    - 색상은 default로 기존에 지정되지 않은 색으로 자동 생성된다. (like, 깃허브 라벨)
- 역량 관리 시에는 역량을 수정할 수 있다.
- 역량 관리 시에는 역량을 삭제할 수 있다.
- 리포트 CUD 는 작성자만 할 수 있다.
- 작성자가 아닌 사용자는 R만 할 수 있다.
- 역량관리 탭은 작성자만 CRUD할 수 있따! 예!
    - 마인드맵 만들어 지면 그 때 작성자가 아닌 사용자도 R 할 수 있다!
- 리포트 역량 그래프 가중치는 `개별 / 전체 * 100` 로 구한다.
    - 사용자는 자신이 원하는 숫자를 입력할 수 있다.
    - 1~10사이의 정수로 입력할 수 있다.

- 리포트 조회 기능
- 역량 그래프 기능
    - 값을 받아서 그래프만 그리는 컴포넌트
    - 값을 수정 하는 기능 (가중치, visible 여부)
- 역량 별 학습로그 조회 기능
- 학습로그 검색 및 선택 기능

<img width="982" alt="스크린샷 2021-08-18 오후 6 26 46" src="https://user-images.githubusercontent.com/67677561/129873930-3f62cc6a-1fc4-4dc5-ae12-8dace6adfb95.png">
<img width="974" alt="스크린샷 2021-08-18 오후 6 27 30" src="https://user-images.githubusercontent.com/67677561/129874034-088a4cb4-4c60-47ba-bf99-18078fd5ceb6.png">

**Comment by devhyun637** [comment]
1. 역량관리 페이지
<img width="580" alt="스크린샷 2021-08-18 오후 11 34 22" src="https://user-images.githubusercontent.com/59258239/129917738-05b040b4-2f74-4820-9e2f-e2e53282652e.png">

2. 리포트 페이지
- 리포트 페이지 첫페이지(디폴트 리포트로 보여짐)
<img width="578" alt="스크린샷 2021-08-18 오후 11 34 04" src="https://user-images.githubusercontent.com/59258239/129917930-a5482887-9a64-42e2-8830-d76f91d68fce.png">

- 새 리포트 작성 또는 리포트 수정 페이지
<img width="574" alt="스크린샷 2021-08-18 오후 11 34 10" src="https://user-images.githubusercontent.com/59258239/129918024-c44467d5-8603-4662-a796-cd9fb2400039.png">

- 역량에 해당하는 학습로그 추가하기(모달)
<img width="570" alt="스크린샷 2021-08-18 오후 11 34 15" src="https://user-images.githubusercontent.com/59258239/129918221-7caa6b7f-f555-453b-9a49-f38aa9e06b4a.png">
